File: Algorithms in Python/insertion-sort.py
# Python program for implementation of Insertion Sort 

# Function to do insertion sort 
def insertionSort(arr): 

	# Traverse through 1 to len(arr) 
	for i in range(1, len(arr)): 

		key = arr[i] 

		# Move elements of arr[0..i-1], that are 
		# greater than key, to one position ahead 
		# of their current position 
		j = i-1
		while j >=0 and key < arr[j] : 
				arr[j+1] = arr[j] 
				j -= 1
		arr[j+1] = key 


# Driver code to test above 
# creating an empty list 
arr = [] 

# number of elements as input 
n = int(input("Enter number of elements : ")) 

# iterating till the range 
for i in range(0, n): 
	ele = int(input()) 

	arr.append(ele) # adding the element 
	
print(arr) 

insertionSort(arr) 
print ("Sorted array is:") 
for i in range(len(arr)): 
	print ("%d" %arr[i]) 

File: Algorithms in Golang/InsertionSort.go
//Program for insertion sort

package main

import (
	"fmt"
)

func main() {

	length := 0
	fmt.Println("Enter the number of inputs: ")
	fmt.Scanln(&length)

	fmt.Println("Enter the inputs: ")

	slice := make([]int, length)
	for i := 0; i < length; i++ {
		fmt.Scanln(&slice[i])
	}

	fmt.Println("\nUnSorted array is: \n\n", slice)
	fmt.Println("\nUnSorted array is: \n\n", slice)
	insertionsort(slice)
	fmt.Println("\nSorted array is: \n", slice, "\n")
}

func insertionsort(items []int) {
	var n = len(items)
	for i := 1; i < n; i++ {
		j := i
		for j > 0 {
			if items[j-1] > items[j] {
				items[j-1], items[j] = items[j], items[j-1]
			}
			j = j - 1
		}
	}
}
